### PR TITLE
Switch to Poetry, adopt tox and pytest

### PR DESCRIPTION
**Pull request by ppfeister**
~~Blocked by #2111~~ Partially consolidated. (Poetry review notes in original blocker, #2111)

`tox` now suitable (and recommended) for running unit tests across several environments at once with linting and coverage reporting.

Will be adopted by GitHub Actions later to remedy lost functionality from Poetry adoption.

### CODE REVIEW

**Comment by ppfeister** [comment]
New PR action set up (also has better coverage)
Action also set to run for each push to master

Will address scheduled actions later.

And for @sdushantha :
`test_validate_manifest_against_remote_schema` validates the data.json in the pushed commitish against the remote schema, and will fail if it doesn't match. This would alert to situations like what happened the other week, where the data.json broke existing clients.

**Comment by ppfeister** [comment]
Windows has been added back with path normalization and Mac has been added with sorting the problematic assertion.

Tests on Python 3.8-3.12 on Linux, Windows, and Mac.

File: docs/install.md [hunk 1]
```diff
@@ -37,13 +37,24 @@ Python
...
 ```bash
-# pipx is recommended, but pip may suffice if pipx is unavailable
-git clone https://github.com/sherlock-project/sherlock.git
-cd sherlock
-pipx install .
+# Assumes repository cloned, and pwd is repository root
+pipx install -e .
+```
+
+### Run package from source (no install)
```
Comment (sdushantha) [inline docs/install.md:51]:
```diff
- ### Run package from source (no install)
+ ### Run package from source (without installing)
```

File: docs/install.md [hunk 2]
```diff
@@ -37,13 +37,24 @@ Python
...
+
+If you'd rather not install directly to your system, you can import the module at runtime with `-m`.
+
+```bash
+# Assumes repository cloned, and pwd is repository root
+python3 -m sherlock user123 user789
```
Comment (sdushantha) [inline docs/install.md:57]: `-m` is not needed. `python sherlock user123 user789` also works

**Review by sdushantha — CHANGES_REQUESTED** [review]

File: docs/install.md (hunk 1 above)
Comment (ppfeister) [reply docs/install.md]: Done

File: docs/install.md (hunk 2 above)
Comment (ppfeister) [reply docs/install.md:57]: That changes with #2111 due to more-proper packaging with Poetry. Leads to import errors and such.

I may be able to address that, but it's not the typical way packages are ran :man_shrugging:

Essentially, when ran as a script, there isn't a proper namespace, and imports have to be relative to the local. When packaged, there is a namespace, and imports have to be self-package-based. I'm explaining this terribly but it's early/late

**Comment by ppfeister** [comment]
Rebased to #2111 feature/pyproj to fix conflicts
Can be merged as a standalone to include both changes or one after another for the same result

File: pytest.ini [hunk 3]
```diff
@@ -0,0 +1,4 @@
+[pytest]
+addopts = --strict-markers
+markers =
+    online: mark tests are requiring interest access.
```
Comment (sdushantha) [inline pytest.ini:4]: Did you mean "internet" or "interest"?

File: docs/CONTRIBUTING.md [hunk 4]
```diff
@@ -68,7 +68,7 @@ Sherlock.  This invocation hides the progress text that Sherlock normally
 outputs, and instead shows the verbose output of the tests.
 
 ```console
-$ cd sherlock/sherlock
+# Assumes pwd is respository root
```
Comment (sdushantha) [inline docs/CONTRIBUTING.md:71]:
```suggestion
# Assumes current working directory is respository root
```

File: docs/INSTALL.md [hunk 5]
```diff
@@ -37,13 +37,24 @@ Python
...
+Building an editable (or live) package links the entry point to your current directory, rather than to the standard install location. This is often useful when working with the code base, as changes are reflected immediately without reinstallation.
+
+Note that the version number will be 0.0.0 for pipx local builds unless manually changed in the pyproject file (it will prompt the user for an update).
 
 ```bash
-# pipx is recommended, but pip may suffice if pipx is unavailable
-git clone https://github.com/sherlock-project/sherlock.git
-cd sherlock
-pipx install .
+# Assumes repository cloned, and pwd is repository root
```
Comment (sdushantha) [inline docs/INSTALL.md:47]:
```suggestion
# Assumes repository cloned, and current working directory is repository root
```

File: docs/INSTALL.md [hunk 6]
```diff
@@ -37,13 +37,24 @@ Python
...
+pipx install -e .
+```
+
+### Run package from source (without installing)
+
+If you'd rather not install directly to your system, you can import the module at runtime with `-m`.
+
+```bash
+# Assumes repository cloned, and pwd is repository root
```
Comment (sdushantha) [inline docs/INSTALL.md:56]:
```suggestion
# Assumes repository cloned, and current working directory is repository root
```

**Review by sdushantha — CHANGES_REQUESTED** [review]

File: pytest.ini (hunk 3 above)
Comment (ppfeister) [reply pytest.ini:4]: Agh. Good catch. Will resolve.

File: docs/CONTRIBUTING.md (hunk 4 above)
Comment (ppfeister) [reply docs/CONTRIBUTING.md:71]: Acceptable rephrase

**Review by sdushantha — APPROVED** [review]

**Pull request merged**